src/common/src/main/scala/osmesa/common/sources/ChangeSource.scala

```diff
@@ -19,47 +18,50 @@ import scala.xml.XML
 object ChangeSource extends Logging {
   val Delay: Duration = 15 seconds
 
-  def getSequence(baseURI: URI, sequence: Int): Seq[Element] = {
+  def getSequence(baseURI: URI, sequence: Int): Seq[Change] = {
     val s = f"$sequence%09d".toArray
     val path =
       s"${s.slice(0, 3).mkString}/${s.slice(3, 6).mkString}/${s.slice(6, 9).mkString}.osc.gz"
 
     logInfo(s"Fetching sequence $sequence")
-    val response =
-      Http(baseURI.resolve(path).toString).asBytes
-
-    if (response.code === 404) {
-      logInfo(s"$sequence is not yet available, sleeping.")
-      Thread.sleep(Delay.toMillis)
-      getSequence(baseURI, sequence)
-    } else {
-      // NOTE: if diff bodies get really large, switch to a SAX parser to help with the memory footprint
-      val bais = new ByteArrayInputStream(response.body)
-      val gzis = new GZIPInputStream(bais)
-      try {
-        val data = XML.loadString(IOUtils.toString(gzis))
-
-        val changes = (data \ "_").flatMap { node =>
-          val action = node.label match {
-            case "create" => Actions.Create
-            case "modify" => Actions.Modify
-            case "delete" => Actions.Delete
+
+    try {
+      val response =
+        Http(baseURI.resolve(path).toString).asBytes
+
+      if (response.code == 404) {
+        logInfo(s"$sequence is not yet available, sleeping.")
+        Thread.sleep(Delay.toMillis)
+        getSequence(baseURI, sequence)
+      } else {
+        // NOTE: if diff bodies get really large, switch to a SAX parser to help with the memory footprint
+        val bais = new ByteArrayInputStream(response.body)
+        val gzis = new GZIPInputStream(bais)
+        try {
+          val data = XML.loadString(IOUtils.toString(gzis))
+
+          val changes = (data \ "_").flatMap { node =>
+            val action = node.label match {
+              case "create" => Actions.Create
+              case "modify" => Actions.Modify
+              case "delete" => Actions.Delete
+            }
+            (node \ "_").map(Change.fromXML(_, action, sequence))
           }
-          (node \ "_").map(Element.fromXML(_, action))
-        }
 
-        logDebug(s"Received ${changes.length} changes")
-
-        changes
-      } catch {
-        case e: IOException =>
-          logWarning(s"Error reading change $sequence", e)
-          Thread.sleep(Delay.toMillis)
-          getSequence(baseURI, sequence)
-      } finally {
-        gzis.close()
-        bais.close()
+          logDebug(s"Received ${changes.length} changes")
+
+          changes
+        } finally {
```

Comment:
These try-with-resources blocks should be fixed, per https://medium.com/@dkomanov/scala-try-with-resources-735baad0fd7d